### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by Vivatist** [shadowsocks/shadowsocks-org#249] [Q&A] [answered]
Hello people. Does SS have some kind of API? For example, I need to see the number of active sessions or some minimal statistics. tnx

**Comment by neteroster** [comment, marked as answer]
For server side, consider using some implementations with metrics such as [outline](https://github.com/Jigsaw-Code/outline-ss-server).

For client side, [clash](https://github.com/Dreamacro/clash) provides you the ability to [view and manage connections](https://dreamacro.github.io/clash/runtime/external-controller.html#connections). If you need an UI, check the [WebUI for clash](https://github.com/Dreamacro/clash-dashboard). If not-opensource is ok for you, [Clash for Windows](https://github.com/Fndroid/clash_for_windows_pkg) can also be considered.

**Comment by Vivatist** [comment]
Tnx!
